src/core/meshcop/meshcop_tlvs.hpp

```diff
@@ -511,7 +511,7 @@ class NetworkNameTlv : public Tlv, public TlvInfo<Tlv::kNetworkName>
      * @retval FALSE  If the TLV does not appear to be well-formed.
      *
      */
-    bool IsValid(void) const { return true; }
+    bool IsValid(void) const;
```

Comment:
Is this called everywhere the NetworkNameTlv is constructed?

Reply:
I added the check to scan results. This is already being called in most places. There are some cases where this isn't needed, e.g., initializing a `NetworkNameTlv` directly from `Mac::GetNetworkName`.

Reply:
Thank you!